__tests__/app.test.js

```diff
@@ -38,61 +36,86 @@ describe("/api/topics", () => {
     return request(app)
-    .get('/api/articles/3')
-    .expect(200)
-    .then(({body})=>{
-      expect(body.article.article_id).toBe(3)
-      expect(body.article).toMatchObject({
-        title: "Eight pug gifs that remind me of mitch",
-        topic: "mitch",
-        author: "icellusedkars",
-        body: "some gifs",
-       created_at: expect.any(String),
-       votes: expect.any(Number),
-        article_img_url:
-          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
-      })
-    })
-  })
-  test('ERROR: 404 responds with an error when a non-existent id is given', ()=>{
+      .get("/api/articles/3")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.article_id).toBe(3);
+        expect(body.article).toMatchObject({
+          title: "Eight pug gifs that remind me of mitch",
+          topic: "mitch",
+          author: "icellusedkars",
+          body: "some gifs",
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
+  test("ERROR: 404 responds with an error when a non-existent id is given", () => {
     return request(app)
-    .get('/api/articles/89')
-    .expect(404)
-    .then(({body})=>{
-      expect(body.msg).toBe('no article with this ID')
-    })
-  })
-  test('ERROR: 400 responds with an error when given an invalid id', ()=>{
+      .get("/api/articles/89")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("no article with this ID");
+      });
+  });
+  test("ERROR: 400 responds with an error when given an invalid id", () => {
+    return request(app)
+      .get("/api/articles/numbers-only")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
+
+describe("/api/articles", () => {
+  test("GET: 200 retrieves all the articles", () => {
     return request(app)
-    .get('/api/articles/numbers-only')
-    .expect(400)
-    .then(({body})=>{
-      expect(body.msg).toBe('Bad Request')
-    })
-  })
-})
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+       const {articles} = response.body
+        expect(articles).toHaveLength(13);
+        expect(articles).toBeSortedBy('created_at', {descending: true})
+        articles.forEach((piece) => {
+          expect(piece).toMatchObject({
+            article_id: expect.any(Number),
+            author: expect.any(String),
+            title: expect.any(String),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+          });
+          expect(piece).hasOwnProperty('comment_count')
+        });
```

Comment:
You should also check the object don't have the `body` property, as these articles could have a key of body and still pass this test. 
Should also have a test that checks the comments are in the correct order (i.e. descending by date/created_at), I recommend `jest-sorted` for this

Reply:
Hi Kathrine, the sorted check is on line106